lib/autoload/send_file.rb

```diff
@@ -0,0 +1,20 @@
+module SendFile
+  # Makes the given file publicly accessible.
+  #
+  # @param [String] file The path to the local file.
+  # @param [String] folder_name The name of the working folder, this name should be unique.
+  # @param [String] file_name The name of public visible file.
+  # @return [String] The url to the public visible file, with the format of
+  #   "/downloads/#{folder_name}/#{file_name}"
+  def publish_file(file, folder_name, file_name)
```

Comment:
Actually filename can be optional.  But it's supposed to be a human readable name. 

e.g. If you have specified a file `"/path_of_the_file/download_&%H#E$.txt"` on disk, the name is random and not human readable. In this case you can specify a human readable name in `file_name`, so that when user get the file, it's properly named.